CONTRIBUTING.md

```diff
@@ -1,8 +1,78 @@
+- Make sure your changes follow the Acceptance Criteria below.
+- When you are confident about the changes, create a Pull Request pointing to the master branch of this repository. 
+
+## Acceptance Criteria ##
+
+Plugins must adhere to a number of conventions, mostly in the way the plugin files are laid out. This allows tools such as the `artifactory-user-plugins-devenv` and our internal testing system to properly recognize and work with plugins. These conventions are as follows:
+
+- Plugin files must be placed inside a plugin folder. The plugin groovy file and its folder must share the same name. The plugin folder can be placed inside a category folder to group plugins under the same subject.
+- If the plugin depends on an external file that it accesses while running, such as a configuration file, this file must have the same name as the plugin groovy file.
+- Update the `README.md` file to be consistent with the changes. If you are proposing a new plugin or if the `README.md` file is missing, use the template below to create a new one.
+- Provide at least one automated test case. If you cannot provide working test cases, use the template below to add an always-fail test case to your plugin. This will make it easier for us to identify plugins that lack test cases and need our attention.
+- Test case groovy files must have the *Test* suffix in their names, and must be capitalized. The name of the test file must match the name of the test class it defines. Test cases should be written using the [Spock framework](http://spockframework.org/spock/docs/1.1/spock_primer.html).
+- If there is exactly one test groovy file and no supplimentary files required to run the test, the file can be in the same folder as the plugin groovy file, and must have the same name as the plugin.
+- If there are multiple test groovy files and/or tests require supplimentary files, all test-related files must be in the `test` subfolder of the plugin folder.
+
+For example, the Artifact Cleanup plugin:
+```
+cleanup/artifactCleanup/README.md                   # The README.md file
+cleanup/artifactCleanup/artifactCleanup.groovy      # The plugin itself
+cleanup/artifactCleanup/artifactCleanup.properties  # An optional configuration file
+cleanup/artifactCleanup/ArtifactCleanupTest.groovy  # The test file
+```
+
+Another example, the Whitesource plugin:
+```
+governance/whitesource/README.md                    # The README.md file
+governance/whitesource/whitesource.groovy           # The plugin itself
+governance/whitesource/whitesource.properties       # An optional configuration file
+governance/whitesource/test/WhitesourceTest.groovy  # The test file
+governance/whitesource/test/guava-15.0.jar          # A test suppliment: this file is deployed to Artifactory as part of the test
+```
+
+## README.md Template ##
```

Comment:
Maybe we should make a template directory with the always-fail testcase, the README.md template, and the dummy plugin?  I know I always start a new plugin by copying the directory of a previous one, and this could be a better approach.

Reply:
@markgalpin we would need to find a way to ignore this plugin in the test automation, since it will always fail. Maybe we could keep it as it for this moment and discuss with Stanley how we can do that.